graph/Graph/Graph.cpp

```diff
@@ -0,0 +1,93 @@
+//
+// Created by emyrmak on 7/9/2019.
+//
+#include <string>
+#include <iostream>
+#include "Graph.h"
+
+Graph::Graph(int& size_t, std::vector<std::vector<int>>& graph) : size(size_t) {
+    matrix = new int*[size];
+    visited = new bool[size];
+    distance = new int[size];
+    way = new std::string[size];
+    for (int i = 0; i < size; i++){
+        matrix[i] = new int[size];
+    }
+    for (int i = 0; i < size; i++) {
+        for (int j = 0; j < size; j++) {
+            matrix[i][j] = 0;
+        }
+    }
+    for (int i = 0; i < graph.size(); i++){
+        matrix[graph[i][0]][graph[i][1]] = graph[i][2];
+    }
+}
+
+void Graph::print() {
+    for (int i = 0; i < size; i++) {
+        for (int j = 0; j < size; j++) {
+            std::cout << matrix[i][j] << " ";
+        }
+        std::cout << std::endl;
+    }
+}
+void Graph::go() {
+    if (readStartEndVertex()){
+        getShortestWay();
+    }
+}
+
+void Graph::getShortestWay(){
+
+    for (int i = 0; i < size; i++){
+         visited[i] = false;
+         distance[i] = infinity;
+         way[i] = "";
+     }
+     distance[startVertex] = 0;
+     way[startVertex] = std::to_string(startVertex);
+     for (int vertex = 0; vertex < size - 1; vertex++){
+         int minDistance = infinity;
+         int currentVertex = 0;
+         for (int i = 0; i < size; i++){
+             if (!visited[i] && distance[i] < minDistance){
+                 currentVertex = i;
+                 minDistance = distance[i];
+             }
+         }
+         visited[currentVertex] = true;
+         for (int i = 0; i < size; i++){
+             if (!visited[i] && matrix[currentVertex][i] && distance[currentVertex]!= infinity &&
```

Comment:
Too long condition. I do not have any idea, what you intend to check. better is to create function like:
bool is_some_condition_fullfiled() 
and put the condition inside